yjit/src/core.rs

```diff
@@ -1881,7 +1899,9 @@ pub fn gen_branch(
 
     // Call the branch generation function
     asm.mark_branch_start(&branchref);
-    gen_fn(asm, branch.dst_addrs[0].unwrap(), branch.dst_addrs[1], BranchShape::Default);
+    if let Some(dst_addr) = branch.dst_addrs[0] {
+        gen_fn(asm, dst_addr, branch.dst_addrs[1], BranchShape::Default);
+    }
```

Comment:
Can you explain why this change?

Reply:
When YJIT runs out of space, it fails to generate the destination code and thus `branch.dst_addrs` becomes `None`. Currently `unwrap()` panics here for that case. It was a blocker to test the current implementation.